pkg/resources/requests.go

```diff
@@ -58,7 +58,9 @@ func (r Requests) ScaledDown(f int64) Requests {
 
 func (r Requests) Divide(f int64) {
 	for k := range r {
-		r[k] /= f
+		if r[k] != 0 {
```

Comment:
Should we check if `f` is zero?

Reply:
This would work either, I just propose this because think that skipping dividing by 0 in general could cover some subtle bugs in the future. 

 I let myself to skip only knowing the resources are 0 as they were earlier in the flow multiplied by 0.

Reply:
I know that is the point, I think it is ok to panic in the low level function if called with r_k >0 and f=0. If we skipped on this case we may in the future introduce subtle bugs.

Reply:
Are we saying that there shouldn't be ever the case with `f = 0`? 
If so, is it being "enforced/checked" else where?

Just of the top: [ComputeTASNetUsage](https://github.com/kubernetes-sigs/kueue/blob/main/pkg/scheduler/flavorassigner/flavorassigner.go#L80)->`ScaleDown`->`Divide`
I don't see any assertions there for `psa.Count` not being `0`

Reply:
When we use ScaleDown, then the original resources were already computed for multiple Pods (multiplied by count=0), and so are 0. 

So, in the crash scenario we have r[k] ==0 and f==0. I prefer to skip only this case, restructured the check adding the check explicitly. ptal